### PR TITLE
Support latest Istio API in profile controller

### PR DESCRIPTION
**Pull request by alembiewski**
Fixes [D2IQ-91529](https://jira.d2iq.com/browse/D2IQ-91529).
Resolves the issue with `EnvoyFilter` on Istio 1.14.1.

### CODE REVIEW

File: components/profile-controller/config/manager/kustomization.yaml [hunk 1]
```diff
@@ -3,9 +3,9 @@ resources:
...
 
 configMapGenerator:
-- name: config
-  literals:
+- literals:
   - ADMIN=
   - WORKLOAD_IDENTITY=
   - USERID_HEADER="kubeflow-userid"
   - USERID_PREFIX=
+  name: config
```
Comment (MessKon) [inline components/profile-controller/config/manager/kustomization.yaml:11]: :newline char needed

Comment (alembiewski) [reply components/profile-controller/config/manager/kustomization.yaml:11]: this was cherry-picked as-is from the upstream commit, generally, I would agree, but in this situation I would propose to keep the changeset as small as possible

File: components/profile-controller/config/crd/patches/trivial_conversion_patch.yaml [hunk 2]
```diff
@@ -3,5 +3,6 @@ kind: CustomResourceDefinition
 metadata:
   name: profiles.kubeflow.org
 spec:
+  preserveUnknownFields: false # TODO: Remove in Kubeflow 1.7 release
   conversion:
     strategy: None
```
[the remark this answers is not in the document]
Comment (alembiewski) [reply components/profile-controller/config/crd/patches/trivial_conversion_patch.yaml:8]: the same as I explained in my previous comment

File: components/profile-controller/config/manager/kustomization.yaml (hunk 1 above)
Comment (MessKon) [reply components/profile-controller/config/manager/kustomization.yaml:11]: makes sense 👍

**Comment by MessKon** [comment]
@alembiewski overall looks good.
2 questions for my benefit.
1. is this related to [D2IQ-91529](https://jira.d2iq.com/browse/D2IQ-91529)? can we add that for clarity?
2. why did we have to update the custom protobuf reference?

**Comment by alembiewski** [comment]
Thanks for the review, @MessKon! I updated the PR description with the JIRA ticket. As for `go.mod` and `go.sum` files, I haven't updated them manually, I think these changes were introduced as a result of `go mod` command during the build

**Review by MessKon — APPROVED** [review]
LGTM ✅

**Pull request merged**